roles/pulp_common/tasks/install_pip.yml

```diff
@@ -181,6 +181,16 @@
         clients: "{{ pulp_install_dir }}/bin/pip"
       register: pip_pkgs
 
+    # Needed because dynaconf bundles box, but dynaconf<3.1.1 conflicts with normally installed
+    # box.
+    - name: Upgrade dynaconf to 3.1.1rc2 pre-release if it is older than 3.1.1
+      pip:
+        name: dynaconf==3.1.1rc2
+        state: present
+        virtualenv: '{{ pulp_install_dir }}'
+        virtualenv_command: '{{ pulp_python_interpreter }} -m venv'
+      when: pip_pkgs.packages[pulp_install_dir + '/bin/pip'].dynaconf[0].version is version("3.1.1", "<")
```

Comment:
I just noticed it now, I didn't know about this `version` plugin.
Nice!

Reply:
Yeah, the ansible docs are quite explicit that you should use it to safely compare versions. A simple numerical operator is unlikely to compare 3.3.3 to 3.3 successfully, the former has 2 dots and is therefore not a float. Or 3.10 to 3.1 .
https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison